frontend/app/src/components/Sidebar/Sidebar.tsx

```diff
@@ -67,68 +67,100 @@ export interface SidebarProps {
+    cachedSidebarWidth || MIN_WIDTH
+  )
+  const [lastInnerWidth, setLastInnerWidth] = useState<number>(
+    window.innerWidth
+  )
 
-  static readonly minWidth = "336"
+  // When hovering sidebar header
+  const [showSidebarCollapse, setShowSidebarCollapse] =
+    useState<boolean>(false)
 
-  public static calculateMaxBreakpoint(value: string): number {
-    // We subtract a margin of 0.02 to use as a max-width
-    return parseInt(value, 10) - 0.02
-  }
+  useEffect(() => {
+    const checkMobileOnResize = (): boolean => {
+      if (!window) return false
 
-  private sidebarRef = React.createRef<HTMLDivElement>()
+      const { innerWidth } = window
 
-  constructor(props: SidebarProps) {
-    super(props)
-    this.mediumBreakpointPx = Sidebar.calculateMaxBreakpoint(
-      props.theme.breakpoints.md
-    )
+      // Collapse the sidebar if the window was narrowed and is now mobile-sized
+      if (innerWidth < lastInnerWidth && innerWidth <= mediumBreakpointPx) {
+        setCollapsedSidebar(true)
+      }
+      setLastInnerWidth(innerWidth)
 
-    const cachedSidebarWidth = localStorageAvailable()
-      ? window.localStorage.getItem("sidebarWidth")
-      : undefined
+      return true
+    }
 
-    this.state = {
-      collapsedSidebar: Sidebar.shouldCollapse(props, this.mediumBreakpointPx),
-      sidebarWidth: cachedSidebarWidth || Sidebar.minWidth,
-      lastInnerWidth: window ? window.innerWidth : Infinity,
-      showSidebarCollapse: false,
+    const handleClickOutside = (event: any): void => {
+      if (sidebarRef && window) {
+        const { current } = sidebarRef
+        const { innerWidth } = window
+
+        if (
+          current &&
+          !current.contains(event.target) &&
+          innerWidth <= mediumBreakpointPx
+        ) {
+          setCollapsedSidebar(true)
+        }
+      }
     }
-  }
 
-  componentDidUpdate(prevProps: any): void {
-    this.mediumBreakpointPx = Sidebar.calculateMaxBreakpoint(
-      this.props.theme.breakpoints.md
-    )
-    // Immediately expand/collapse sidebar when initialSidebarState or hasElements changes
-    if (
-      this.props.initialSidebarState !== prevProps.initialSidebarState ||
-      this.props.hasElements !== prevProps.hasElements
-    ) {
-      this.setState({
-        collapsedSidebar: Sidebar.shouldCollapse(
-          this.props,
-          this.mediumBreakpointPx
-        ),
-      })
+    window.addEventListener("resize", checkMobileOnResize)
+    document.addEventListener("mousedown", handleClickOutside)
+
+    return () => {
+      window.removeEventListener("resize", checkMobileOnResize)
+      document.removeEventListener("mousedown", handleClickOutside)
     }
-  }
+  }, [lastInnerWidth, mediumBreakpointPx])
 
-  static shouldCollapse(
-    props: SidebarProps,
+  useEffect(() => {
+    setCollapsedSidebar(
+      shouldCollapse(initialSidebarState, mediumBreakpointPx)
+    )
+  }, [initialSidebarState, hasElements, mediumBreakpointPx])
```

Comment:
If `hasElements` is indeed needed here in the deps array, could you please add a comment why, since it is not used in the `useEffect` itself?